app/models/upload.rb

```diff
@@ -16,19 +17,9 @@ def can_calculate
     %w(cat1_r2 cat1_r3 cat1_r31 cat1_r4).include?(file_type)
   end
 
-  def completed?
-    state == :complete || state == :failed
-  end
-
-  def complete(message = '')
-    self.state = :complete
-    self.status_message = message
-    save!(validate: false)
-  end
-
-  def fail(message = '')
+  def fail(message = '', qrda_file = nil)

```

Comment:
Does `qrda_file` get used here?

Reply:
Oops, meant to log which file it was failing on and apparently missed it, fixed!